### PR TITLE
feat: 动画样式添加 prefix

### PR DESCRIPTION
**Pull request by crazyair**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Motion related css className prepend `ant-` prefix to avoid naming conflict.      |
| 🇨🇳 Chinese |   动画相关 css class 全部添加 `ant-` 前缀以防止重名问题。       |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Review by xrkffgg — COMMENTED** [review]
feature branch

**Comment by crazyair** [comment]
> feature branch

也算修复 bug，现在问题是 v3，v4 混用，样式有冲突

**Comment by zombieJ** [comment]
> 也算修复 bug，现在问题是 v3，v4 混用，样式有冲突

breaking change 发 feature

**Comment by xrkffgg** [comment]
rebase feature 
push -f

**Comment by crazyair** [comment]
- message  `prefixCls` => `rootPrefixCls `
- `transitionName`  参数强制添加 prefix，用户自定义也包含 `ant-` 前缀
- `motionName` 需要加 prefix 吗？

File: components/dropdown/__tests__/index.test.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ describe('Dropdown', () => {
 
   it('overlay is function and has custom transitionName', () => {
     const wrapper = mount(
-      <Dropdown overlay={() => <div>menu</div>} transitionName="move-up" visible>
+      <Dropdown overlay={() => <div>menu</div>} transitionName="ant-move-up" visible>
```
Comment (zombieJ) [inline components/dropdown/__tests__/index.test.js:23]: 这个测试不用改，本身就是测自定义的

Comment (crazyair) [reply components/dropdown/__tests__/index.test.js:23]: 源码改的是，自定义传的也强制加 prefix 
image

**Review by zombieJ — APPROVED** [review]

**Comment by zombieJ** [comment]
changelog 加一下就准备合了~

**Comment by zombieJ** [comment]
https://preview-29268-ant-design.surge.sh/components/select-cn/  Select 下拉动画没了

**Comment by crazyair** [comment]
@shaodahong https://github.com/react-component/image/blob/39ab219ec9fa12dbb8c96df9ac9c46896d6018fb/src/Preview.tsx#L238-L239 Image 组件现在有问题，没法设置 prefix。

**Comment by shaodahong** [comment]
> @shaodahong https://github.com/react-component/image/blob/39ab219ec9fa12dbb8c96df9ac9c46896d6018fb/src/Preview.tsx#L238-L239 Image 组件现在有问题，没法设置 prefix。

```jsx
<Image prefixCls="abc" previewPrefixCls="def" />
```

**Pull request merged**